package.json

```diff
@@ -19,7 +19,10 @@
         "mocha": "^7.1.2",
         "mock-require": "^3.0.3",
         "nock": "^12.0.3",
-        "nyc": "^15.0.1"
+        "nyc": "^15.0.1",
+        "semantic-release": "17.0.7",
+        "@semantic-release/git": "9.0.0",
+        "conventional-changelog-eslint": "3.0.8"
```

Comment:
What does `conventional-changelog-eslint` do?

Reply:
Additional commit message parsing?

Reply:
https://www.npmjs.com/package/conventional-changelog-eslint

IIUC, this checks our commit messages to make sure they are understandable by semantic release...aka so we don't accidentally forget to release, etc.

Reply:
semantic-release uses the commit messages to determine the type of changes in the codebase. By default it uses Angular Commit Message Conventions.  - 
Reference: https://github.com/semantic-release/semantic-release#commit-message-format

It lets us configure the commit conventions as a preset, in which case it must be installed.
Reference: https://github.com/semantic-release/commit-analyzer#options

For now, I have kept the current convention very simple. 

`By default for any commit message there will be no release.
Only if someone wants a release, they have to say it in commit message by **prefixing** 
- if they want patch(BUGFIX-RELEASE:) , minor(FEATURE-RELEASE:) or major(BREAKING-RELEASE:)  in the commit message. `